src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -361,21 +361,21 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 		return undefined;
 	}
 
-	public fileToBase64 = (filepath, getType) => {
-		return new Promise<string | ArrayBuffer>(async resolve => {
+	public async getBase64MimeType(filepath: string, getMimeType: boolean): Promise<string> {
```

Comment:
This function name isn't accurate currently - you only get the mime type if `getMimeType` is set to true. And you're not getting a "base64 mime type" (whatever that would mean) - you're just getting the mime type of the blob itself. The base64 stuff doesn't even come into play at that point. 

A better pattern would be to split this into two separate functions `getMimeType` and `getFileContentBase64` or something - that is if there isn't existing functions that already do this or some other simple functionality you could use such as the ones I've suggested in the other comment thread.

Reply:
I changed it from getBase64OrMimeType, I thought it's easy enough for folks to decode that when they look at the parameters and splitting it makes sense if am moving it to a common place.

I don't really love to complicate the code. I tried all the easy ways (including your above suggestions) and none of them worked So I had to do what I did. I would really love to simplify it if you could help it!

Reply:
Splitting it into two functions would be a good start - they're getting very different values and so having them be in the same function is just confusing.